database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -255,6 +252,9 @@ public void onDataChange(DataSnapshot snapshot) {
                     Log.w(TAG, "Key not found at ref: " + snapshot.getRef());
                 }
             }
+
+            mKeysWithPendingUpdate.remove(key);
```

Comment:
Previously we only executed these two lines if:

- Snapshot had a non-null value AND
- `isKeyAtIndex() matched`

Now we are executing it regardless of snapshot data.  Can you add a comment explaining what we are doing here and when it matters?